### PR TITLE
update tiltfile to support azure cni dev

### PR DESCRIPTION
**Pull request by nawazkh**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
- Updates Tiltfile for local development of workload cluster using Azure CNI v1

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/3610

**Special notes for your reviewer**:
- PR https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3508 should be merged before merging this PR.

- [ ] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nawazkh** [comment]
Was able to bring up workload cluster using Azure cni v1 via Tilt.
Screenshot 2023-06-14 at 4 50 03 PM

Screenshot 2023-06-14 at 4 50 12 PM

File: Tiltfile [hunk 1]
```diff
@@ -382,21 +382,24 @@ def get_addons(flavor_name):
...
-        calico_values = "./templates/addons/calico-dual-stack/values.yaml"
-    else:
-        calico_values = "./templates/addons/calico/values.yaml"
-
-    addon_cmd = "; " + helm_cmd + " --kubeconfig ./${CLUSTER_NAME}.kubeconfig install --repo https://docs.tigera.io/calico/charts --version ${CALICO_VERSION} calico tigera-operator -f " + calico_values + " --namespace tigera-operator --create-namespace"
-
+    addon_cmd = ""
     if "intree-cloud-provider" not in flavor_name:
-        addon_cmd += "; " + helm_cmd + " --kubeconfig ./${CLUSTER_NAME}.kubeconfig install --repo https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/helm/repo cloud-provider-azure --generate-name --set infra.clusterName=${CLUSTER_NAME}"
+        addon_cmd += "; " + helm_cmd + " --kubeconfig ./${CLUSTER_NAME}.kubeconfig install --repo https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/helm/repo cloud-provider-azure --generate-name --set infra.clusterName=${CLUSTER_NAME} --set cloudControllerManager.clusterCIDR='192.168.0.0/16'"
```
Comment (CecileRobertMichon) [inline Tiltfile:387]: Is there a way we could avoid hardcoding `cloudControllerManager.clusterCIDR` to `192.168.0.0/16` by setting it dynamically based on the actual value of the `Cluster`?

Comment (nawazkh) [reply Tiltfile]:
```
addon_cmd += "; export CIDRS=$(" + kubectl_cmd + " get cluster ${CLUSTER_NAME} -o jsonpath='{.spec.clusterNetwork.pods.cidrBlocks[*]}')"
addon_cmd += "; export CIDR_LIST=$(bash -c 'echo $CIDRS' | tr ' ' ',')"
```

Should now get `CIDR_LIST` and add to the helm command dynamically.

**Comment by CecileRobertMichon** [comment]
@nawazkh I tried deploying a cluster with the "azure-cni-v1" flavor on tilt, everything came up successfully and nodes are Ready but when I list the pods I can see the coredns pods aren't ready:

```
k get pods -A                                          
NAMESPACE     NAME                                                             READY   STATUS    RESTARTS      AGE
kube-system   azure-cni-549kt                                                  1/1     Running   0             5m5s
kube-system   azure-cni-gbxld                                                  1/1     Running   0             4m25s
kube-system   azure-cni-zlqnx                                                  1/1     Running   0             6m34s
kube-system   cloud-controller-manager-749955c9db-fgkmw                        1/1     Running   0             6m35s
kube-system   cloud-node-manager-lcmgk                                         1/1     Running   0             6m35s
kube-system   cloud-node-manager-qkc62                                         1/1     Running   0             5m5s
kube-system   cloud-node-manager-xdb47                                         1/1     Running   0             4m25s
kube-system   coredns-787d4945fb-4cx72                                         0/1     Running   3 (51s ago)   7m11s
kube-system   coredns-787d4945fb-g88bt                                         0/1     Running   3 (51s ago)   7m11s
kube-system   etcd-azure-cni-v1-14036-control-plane-bkg82                      1/1     Running   0             7m10s
kube-system   kube-apiserver-azure-cni-v1-14036-control-plane-bkg82            1/1     Running   0             7m11s
kube-system   kube-controller-manager-azure-cni-v1-14036-control-plane-bkg82   1/1     Running   0             7m10s
kube-system   kube-proxy-66ps8                                                 1/1     Running   0             5m5s
kube-system   kube-proxy-bjkfz                                                 1/1     Running   0             7m11s
kube-system   kube-proxy-z28mj                                                 1/1     Running   0             4m25s
kube-system   kube-scheduler-azure-cni-v1-14036-control-plane-bkg82            1/1     Running   0             7m10s
```

Looking at the pod describe I see 

>   Warning  Unhealthy         54s (x44 over 5m57s)   kubelet            Readiness probe failed: Get "http://10.0.0.45:8181/ready": dial tcp 10.0.0.45:8181: connect: connection refused

**Comment by CecileRobertMichon** [comment]
Was able to get this working with k8s v1.27.2. The above repro uses k8s v1.26.1.

**Comment by nawazkh** [comment]
> Was able to get this working with k8s v1.27.2. The above repro uses k8s v1.26.1.

I dont see this behavior using K8s `v1.26.6`
```
Every 1.0s: kubectl get pods -A -o wide                                                                                                                                                                           MacBook-Pro-2.local: Tue Jul 11 12:45:58 2023

NAMESPACE     NAME                                                            READY   STATUS    RESTARTS   AGE   IP           NODE                                    NOMINATED NODE   READINESS GATES
kube-system   azure-cni-h72bl                                                 1/1     Running   0          13m   10.1.0.114   azure-cni-v1-3096-md-0-cxwvv            <none>           <none>
kube-system   azure-cni-ndg5n                                                 1/1     Running   0          15m   10.0.0.4     azure-cni-v1-3096-control-plane-tgmkb   <none>           <none>
kube-system   azure-cni-qbnwt                                                 1/1     Running   0          13m   10.1.0.4     azure-cni-v1-3096-md-0-4rq66            <none>           <none>
kube-system   cloud-controller-manager-5757c65db4-5dmpt                       1/1     Running   0          15m   10.0.0.4     azure-cni-v1-3096-control-plane-tgmkb   <none>           <none>
kube-system   cloud-node-manager-6ptkl                                        1/1     Running   0          15m   10.0.0.4     azure-cni-v1-3096-control-plane-tgmkb   <none>           <none>
kube-system   cloud-node-manager-86f9x                                        1/1     Running   0          13m   10.1.0.4     azure-cni-v1-3096-md-0-4rq66            <none>           <none>
kube-system   cloud-node-manager-9bqkw                                        1/1     Running   0          13m   10.1.0.114   azure-cni-v1-3096-md-0-cxwvv            <none>           <none>
kube-system   coredns-787d4945fb-r9w22                                        1/1     Running   0          16m   10.0.0.33    azure-cni-v1-3096-control-plane-tgmkb   <none>           <none>
kube-system   coredns-787d4945fb-rnrq6                                        1/1     Running   0          16m   10.0.0.7     azure-cni-v1-3096-control-plane-tgmkb   <none>           <none>
kube-system   etcd-azure-cni-v1-3096-control-plane-tgmkb                      1/1     Running   0          16m   10.0.0.4     azure-cni-v1-3096-control-plane-tgmkb   <none>           <none>
kube-system   kube-apiserver-azure-cni-v1-3096-control-plane-tgmkb            1/1     Running   0          16m   10.0.0.4     azure-cni-v1-3096-control-plane-tgmkb   <none>           <none>
kube-system   kube-controller-manager-azure-cni-v1-3096-control-plane-tgmkb   1/1     Running   0          16m   10.0.0.4     azure-cni-v1-3096-control-plane-tgmkb   <none>           <none>
kube-system   kube-proxy-7cz7w                                                1/1     Running   0          13m   10.1.0.4     azure-cni-v1-3096-md-0-4rq66            <none>           <none>
kube-system   kube-proxy-qcllx                                                1/1     Running   0          16m   10.0.0.4     azure-cni-v1-3096-control-plane-tgmkb   <none>           <none>
kube-system   kube-proxy-t8bxc                                                1/1     Running   0          13m   10.1.0.114   azure-cni-v1-3096-md-0-cxwvv            <none>           <none>
kube-system   kube-scheduler-azure-cni-v1-3096-control-plane-tgmkb            1/1     Running   0          16m   10.0.0.4     azure-cni-v1-3096-control-plane-tgmkb   <none>           <none>

```

```
❯ k version -o yaml
clientVersion:
  buildDate: "2023-05-17T14:20:07Z"
  compiler: gc
  gitCommit: 7f6f68fdabc4df88cfea2dcf9a19b2b830f1e647
  gitTreeState: clean
  gitVersion: v1.27.2
  goVersion: go1.20.4
  major: "1"
  minor: "27"
  platform: darwin/arm64
kustomizeVersion: v5.0.1
serverVersion:
  buildDate: "2023-06-14T09:49:08Z"
  compiler: gc
  gitCommit: 11902a838028edef305dfe2f96be929bc4d114d8
  gitTreeState: clean
  gitVersion: v1.26.6
  goVersion: go1.19.10
  major: "1"
  minor: "26"
  platform: linux/amd64

```

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**